### ISSUE TITLE
amazon-ebs: [FATAL] failed to allocate memory

### ISSUE THREAD

**Issue by mohamaa** [mitchellh/packer#3256]
I am trying to bootstrap chef-client in packer with a run list. But it always gives following error :

```**
amazon-ebs: [2016-02-24T03:18:04+00:00] INFO: Processing chef_gem[aws-sdk] action install (aws::default line 20)
amazon-ebs: [FATAL] failed to allocate memory
amazon-ebs: Creating configuration file 'knife.rb'
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: No AMIs to cleanup
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' errored: Error executing Chef: Non-zero exit status: 1

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: Error executing Chef: Non-zero exit status: 1

==> Builds finished but no artifacts were created.
```
Also I am setting up the winrm inside builders :

```
"builders": [{
    "type": "amazon-ebs",
    "access_key": "{{user `aws_access_key`}}",
    "secret_key": "{{user `aws_secret_key`}}",
    "region": "us-east-1",
    "source_ami": "ami-xxxxxxxx",
	"user_data_file": "setup_winrm.txt",
    "instance_type": "m3.xlarge",
	"communicator": "winrm",
	"winrm_username": "Administrator",
	"winrm_password": "xxxxxxxxxx",
    "ami_name": "packer-run {{timestamp}}"
  }]
```
where my setup_winrm.txt file content is :

```
<powershell>
# set administrator password
net user Administrator xxxxxxxx
wmic useraccount where "name='Administrator'" set PasswordExpires=FALSE

winrm quickconfig -q
winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="4096"}'
winrm set winrm/config '@{MaxTimeoutms="1800000"}'
winrm set winrm/config/service '@{AllowUnencrypted="true"}'
winrm set winrm/config/service/auth '@{Basic="true"}'

netsh advfirewall firewall add rule name="WinRM 5985" protocol=TCP dir=in localport=5985 action=allow
netsh advfirewall firewall add rule name="WinRM 5986" protocol=TCP dir=in localport=5986 action=allow

net stop winrm
sc config winrm start=auto
net start winrm
Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope LocalMachine
</powershell>
```

My provisioners are :
```

 "provisioners": [
	{
	"type": "powershell",
	"inline": [
	  "(New-Object System.Net.WebClient).DownloadFile('http://chef.io/chef/install.msi', 'C:\\Windows\\Temp\\chef.msi')",
	  "Start-Process 'msiexec' -ArgumentList '/qb /i C:\\Windows\\Temp\\chef.msi' -NoNewWindow -Wait"
	]
	},
	{
	"type": "file",
	"source": "C:\\Users\\mohamaa\\Documents\\Terraform-work\\identity-validator.pem",
	"destination": "C:\\chef\\identity-validator.pem"
	},
	{
	"type": "powershell",
	"inline": ["echo Y| cacls 'C:\\chef' /g everyone:f"]
	},
    {
	"type": "chef-client",
	"guest_os_type": "windows",
	"staging_directory": "C:\\\\chef",
	"server_url": "https://xxxxxx/organizations/identity",
	"skip_clean_node": "true",
	"skip_clean_client": "true",
	"skip_install": "true",
	"validation_client_name": "xxxxxxx-validator",
    "validation_key_path": "C:\\Users\\mohamaa\\Documents\\Terraform-work\\identity-validator.pem",
	"node_name": "packer-splunked-chef-server",
	"chef_environment": "_default",
	"run_list": ["recipe[windows::default]","recipe[aws::default]","recipe[splunk_windows::default]"]
    },
	{
	"type": "powershell",
	"inline": ["Get-ChildItem -Path 'C:\\chef' -Include *.* -File -Recurse | foreach { $_.Delete()}"]
	}]
}
```

To resolve the issue i tried following things:

1. Tried increasing MaxMemoryPerShellMB from 300 to 1024 and then to 2048 and then to 4096
But same memory exception : [FATAL] failed to allocate memory
2. I tried increasing the instance_type to m3.xlarge but no success.
3. I tried removing the recipe[aws::default] from run list and it worked fine.

My requirement is i want recipe[aws::default] to be a part of my run list.

Please help.

versions used:
Using Chef 12.7.2 here.
aws cookbook version 2.9.2
packer v0.9.0

**Comment by cbednarski** [comment]
Thanks for the report! Sorry you've run into a bug here. Just to double-check, are you using packer 0.9? Some of this code is new / has changed significantly in the latest release so if are using a different version can you try with 0.9?

**Comment by mohamaa** [comment]
Yes i am using packer v0.9.0

**Comment by mohamaa** [comment]
Initially i thought it could be an issue with **remote winrm bootstrapping** so i tried both the ways as :

1. Ran **chef-client** from inside the aws windows server with the run list containing **aws::default** cookbook and it worked fine.
2. Tried bootstrapping the aws windows server from my local workstation using knife bootstrap command as :
```
knife bootstrap windows winrm xxx.compute-1.amazonaws.com --winrm-user Administrator
 --winrm-password 'xxx' --node-name test1 -r "recipe[windows::default],recipe[aws::default],recipe[splunk_
windows::default]"
```
It ran without any error.

So its not an issue with remote windows knife bootstrap, but with **Packer/Terraform** it self.

May i request to please help me out in resolving this ?

**Comment by mohamaa** [comment]
The issue got resolved by applying the windows hot fix KB2842230 on aws ami.

**Comment by rickard-von-essen** [comment]
Great to hear! Closing

**Comment by jtstrohl** [comment]
I seem to be having this same issue, but on Windows 2012 R2, and only recently (was working before) -- only the KB2842230 patch is not available for the newer version of Windows Server.  I've tried manually increasing the WinRM MaxMemoryPerShellMB parameter, but to no avail.
```
2018/04/24 15:58:02 packer: 2018/04/24 15:58:02 [INFO] command 'c:/opscode/chef/bin/chef-client.bat --no-color -c c:/chef/client.rb -E packer -j C:/Windows/Temp/packer-chef-client/first-boot.json' exited with code: 1
2018/04/24 15:58:02 packer: 2018/04/24 15:58:02 [INFO] RPC endpoint: Communicator ended with: 1
2018/04/24 15:58:02 [INFO] 0 bytes written for 'stderr'
2018/04/24 15:58:02 [INFO] 12871 bytes written for 'stdout'
2018/04/24 15:58:02 [INFO] RPC client: Communicator ended with: 1
2018/04/24 15:58:02 [INFO] RPC endpoint: Communicator ended with: 1
2018/04/24 15:58:02 packer: 2018/04/24 15:58:02 [INFO] 0 bytes written for 'stderr'
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: ================================================================================[0m
[0;32m    amazon-ebs: ================================================================================[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: Error executing action `install` on resource 'chef_gem[aws-sdk]'[0m
[0;32m    amazon-ebs: Error executing action `install` on resource 'chef_gem[aws-sdk]'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: ================================================================================[0m
[0;32m    amazon-ebs: ================================================================================[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs:[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: Mixlib::ShellOut::ShellCommandFailed[0m
[0;32m    amazon-ebs: Mixlib::ShellOut::ShellCommandFailed[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: ------------------------------------[0m
[0;32m    amazon-ebs: ------------------------------------[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: Expected process to exit with [0], but received '1'[0m
[0;32m    amazon-ebs: Expected process to exit with [0], but received '1'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: ---- Begin output of c:/opscode/chef/embedded/bin/gem install aws-sdk -q --no-rdoc --no-ri -v "3.0.1" ----[0m
[0;32m    amazon-ebs: ---- Begin output of c:/opscode/chef/embedded/bin/gem install aws-sdk -q --no-rdoc --no-ri -v "3.0.1" ----[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: STDOUT:[0m
[0;32m    amazon-ebs: STDOUT:[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: STDERR: c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1482:in `wait_readable': failed to allocate memory (NoMemoryError)[0m
[0;32m    amazon-ebs: STDERR: c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1482:in `wait_readable': failed to allocate memory (NoMemoryError)[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1482:in `begin_transport'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1433:in `transport_request'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1407:in `request'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/request.rb:221:in `perform_request'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/request.rb:156:in `fetch'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/remote_fetcher.rb:368:in `request'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/remote_fetcher.rb:251:in `fetch_http'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/remote_fetcher.rb:292:in `fetch_path'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/source.rb:153:in `fetch_spec'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/index_specification.rb:65:in `spec'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/specification.rb:104:in `installable_platform?'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/platform.rb:35:in `installable?'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:224:in `block in select_local_platforms'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:223:in `select'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:223:in `select_local_platforms'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:214:in `find_possible'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:229:in `search_for'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:10:in `block in search_for'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:69:in `with_no_such_dependency_error_handling'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:9:in `search_for'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:449:in `push_state_for_requirements'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:438:in `require_nested_dependencies_for'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:422:in `activate_spec'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:388:in `attempt_to_activate_new_spec'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1482:in `begin_transport'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1433:in `transport_request'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/2.3.0/net/http.rb:1407:in `request'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/request.rb:221:in `perform_request'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/request.rb:156:in `fetch'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/remote_fetcher.rb:368:in `request'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/remote_fetcher.rb:251:in `fetch_http'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/remote_fetcher.rb:292:in `fetch_path'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/source.rb:153:in `fetch_spec'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/index_specification.rb:65:in `spec'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/specification.rb:104:in `installable_platform?'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/platform.rb:35:in `installable?'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:224:in `block in select_local_platforms'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:223:in `select'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:223:in `select_local_platforms'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:214:in `find_possible'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:229:in `search_for'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:10:in `block in search_for'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:69:in `with_no_such_dependency_error_handling'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/delegates/specification_provider.rb:9:in `search_for'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:449:in `push_state_for_requirements'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:438:in `require_nested_dependencies_for'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:422:in `activate_spec'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:388:in `attempt_to_activate_new_spec'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:319:in `attempt_to_activate'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:319:in `attempt_to_activate'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:137:in `process_topmost_state'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:137:in `process_topmost_state'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:78:in `resolve'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolver.rb:42:in `resolve'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:191:in `resolve'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/request_set.rb:368:in `resolve'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/dependency_installer.rb:484:in `resolve_dependencies'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:194:in `install_gem'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolution.rb:78:in `resolve'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver/molinillo/lib/molinillo/resolver.rb:42:in `resolve'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/resolver.rb:191:in `resolve'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/request_set.rb:368:in `resolve'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/dependency_installer.rb:484:in `resolve_dependencies'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:194:in `install_gem'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:255:in `block in install_gems'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:255:in `block in install_gems'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:251:in `each'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:251:in `each'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:251:in `install_gems'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:251:in `install_gems'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:158:in `execute'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/commands/install_command.rb:158:in `execute'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/command.rb:310:in `invoke_with_build_args'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/command.rb:310:in `invoke_with_build_args'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/command_manager.rb:169:in `process_args'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/command_manager.rb:169:in `process_args'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/command_manager.rb:139:in `run'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/command_manager.rb:139:in `run'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/gem_runner.rb:55:in `run'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/lib/ruby/site_ruby/2.3.0/rubygems/gem_runner.rb:55:in `run'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs:  from c:/opscode/chef/embedded/bin/gem:21:in `<main>'[0m
[0;32m    amazon-ebs:  from c:/opscode/chef/embedded/bin/gem:21:in `<main>'[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: ---- End output of c:/opscode/chef/embedded/bin/gem install aws-sdk -q --no-rdoc --no-ri -v "3.0.1" ----[0m
[0;32m    amazon-ebs: ---- End output of c:/opscode/chef/embedded/bin/gem install aws-sdk -q --no-rdoc --no-ri -v "3.0.1" ----[0m
2018/04/24 15:58:02 ui: [0;32m    amazon-ebs: Ran c:/opscode/chef/embedded/bin/gem install aws-sdk -q --no-rdoc --no-ri -v "3.0.1" returned 1[0m
[0;32m    amazon-ebs: Ran c:/opscode/chef/embedded/bin/gem install aws-sdk -q --no-rdoc --no-ri -v "3.0.1" returned 1
```

**Comment by jtstrohl** [comment]
Figured out my issue..... WinRM has it's own memory cap.  Even though I had raised the cap manually on the server, I realized that we had hard-coded the WinRM MaxMemoryPerShellMB parameter to 1024MB (apparently too restrictive) in the powershell script we used to set up WinRM for Packer remote execution (See example here: http://blog.petegoo.com/2016/05/10/packer-aws-windows/).  This script was overriding the value I set manually.  Once we increased from 1024 to 4096 MB in the WinRM bootstrap script, it started working.
